Create query for user list

pedro-m-dietrich opened the pull request:
The `users` query fetches a list of users, using alphabetical order, with the specified number of users and an offset.

The response contains, besides the user list, the total amount of users in the database, the amount of users in the response, the offset and if it contains the last page (end of the list).

pedro-m-dietrich committed:
Create query for fetching multiple users

A query has been added to fetch a list of users in alphabetical order.
The request requires a JWT authentication.

pedro-m-dietrich committed:
Add pagination to `users` query

When querying for a list of users, it is now possible to specify an
offset. The response contains data about the fetched user list, allowing
pagination.

alanraso commented on src/server.ts:
```diff
@@ -53,6 +58,30 @@ async function getUser(userId: GetUserInput): Promise<User> {
 }
 
+async function getUsers(usersInput: UserListInput): Promise<UserList> {
+  const limit = usersInput?.userLimit ? usersInput.userLimit : 10;
```
```suggestion
  const limit = usersInput?.userLimit ?? 10;
```

alanraso commented on src/server.ts:
```diff
@@ -53,6 +58,30 @@ async function getUser(userId: GetUserInput): Promise<User> {
 
+async function getUsers(usersInput: UserListInput): Promise<UserList> {
+  const limit = usersInput?.userLimit ? usersInput.userLimit : 10;
+  const offset = usersInput?.offset ? usersInput.offset : 0;
```
```suggestion
  const offset = usersInput?.offset ?? 0;
```

Mesmo operador: nullish coalescing.

alanraso commented on src/server.ts:
```diff
@@ -149,6 +178,10 @@ const resolvers = {
     },
+    users: async (_, args: { usersInput: UserListInput }, context): Promise<UserList> => {
+      verifyUserID(context.authResult);
+      return getUsers(args?.usersInput ? args.usersInput : { userLimit: 10, offset: 0 });
```
```suggestion
      return getUsers(args?.usersInput ?? { userLimit: 10, offset: 0 });
```

alanraso commented on src/typedefs.ts:
```diff
@@ -31,6 +44,7 @@ export const typeDefs = `#graphql
   type Query {
     hello: String!
     user(userId: GetUserInput!): User
+    users(usersInput: UserListInput): UserList!
```
É possível você declarar na tipagem do GraphQL que o argumento tem um valor default. Se não me engano é assim:

```ts
    users(usersInput: UserListInput = { offset: 0, limit: 10 }): UserList!
```

Assim o comportamento que o seu código tem de estabelecer esses 2 valores default também se reflete no modelo de tipagem/documentação do GraphQL

pedro-m-dietrich committed:
Refactor default values on `users` query

Default values were added in the GraphQL type definitions for the
`users` query. The code for the query has changed accordingly.

matheusflds commented on src/server.ts:
```diff
@@ -113,14 +141,60 @@ async function insertUserIntoDB(userData: UserInput): Promise<User> {
+  }
+
+  if (user && bcrypt.compareSync(loginInput.password, user.password)) {
+    let expiresTime = '30m';
+    if (loginInput?.rememberMe) {
+      expiresTime = '7d';
+    }
```
nit: 
```suggestion
    const expiresTime = loginInput?.rememberMe ? '7d' : '30m';
```

pedro-m-dietrich committed:
Refactor JWT expiration time for `RememberMe`
